auth.py

```
@@ -106,3 +106,99 @@ def complex_auth_function(user, pass_word, email, phone, address, city, state, z
+def calculate_user_score(total_points, completed_tasks):
+    return total_points / completed_tasks
+
+
+def build_user_report(users):
+    report = ""
+    for user in users:
+        report = report + f"User: {user['name']}, Email: {user['email']}\n"
+    return report
+
+
+def add_user_permissions(user_id, permissions=[]):
```

Ruff (0.12.2)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)